src/transformers/cache_utils.py

```diff
@@ -447,6 +447,110 @@ def batch_select_indices(self, indices: torch.Tensor):
             self.value_cache[layer_idx] = self.value_cache[layer_idx][indices, ...]
 
 
+class OffloadedCache(DynamicCache):
+    """
```

Comment:
at some point we should warn if generating on CPU that GPU is required + if we use cuda stream this requires a hardware that supports cuda streams

Reply:
Hmm, the point of this class is you want to generate on the GPU but you have more CPU RAM than GPU RAM. If you are generating on the CPU you don't need this class at all.

Reply:
Added a check to protect this.